File: frontend/src/api/shop.ts
import { del, downloadBlob, get, post, put, uploadForm } from './index'

export interface Shop {
  id: number
  org_id: number
  org_name?: string | null
  platform_name: string
  shop_name: string
  shop_color?: string
  tax_no?: string
  merchant?: string
  registered_address?: string
  legal_person?: string
  previous_name?: string
  store_long_id?: string
  store_short_id?: string
  settlement_period?: string
  primary_account?: string
  anchor?: string
  shop_type?: string
  purpose?: string
  former_name?: string
  remark?: string
  status: number
  created_at: string
  updated_at?: string
}

export interface ShopForm {
  platform_name: string
  shop_name: string
  shop_color?: string | null
  tax_no?: string | null
  merchant?: string | null
  registered_address?: string | null
  legal_person?: string | null
  previous_name?: string | null
  store_long_id?: string | null
  store_short_id?: string | null
  settlement_period?: string | null
  primary_account?: string | null
  anchor?: string | null
  shop_type?: string | null
  purpose?: string | null
  former_name?: string | null
  remark?: string | null
}

export interface ShopListParams {
  page?: number
  page_size?: number
  keyword?: string
  platform_name?: string
  org_id?: number | string
}

export interface ShopImportError {
  row: number
  message: string
}

export interface ShopImportResult {
  total: number
  created: number
  updated: number
  skipped: number
  errors: ShopImportError[]
}

export function getShopList(params: ShopListParams) {
  return get<{ items: Shop[]; total: number }>('/shops', params as Record<string, any>)
}

export function getShopDetail(id: number) {
  return get<Shop>(`/shops/${id}`)
}

export function createShop(data: ShopForm) {
  return post<Shop>('/shops', data)
}

export function updateShop(id: number, data: Partial<ShopForm> & { status?: number }) {
  return put<Shop>(`/shops/${id}`, data)
}

export function deleteShop(id: number) {
  return del<void>(`/shops/${id}`)
}

export function downloadShopImportTemplate() {
  return downloadBlob('/shops/import-template')
}

export function exportShops(params: ShopListParams & { ids?: string }) {
  return downloadBlob('/shops/export', params as Record<string, any>)
}

export function importShops(file: File) {
  const data = new FormData()
  data.append('file', file)
  return uploadForm<ShopImportResult>('/shops/import', data)
}

File: backend/app/schemas/shop.py
from datetime import datetime

from pydantic import BaseModel, Field


class ShopInfoFields(BaseModel):
    tax_no: str | None = Field(None, max_length=100, description="税号")
    merchant: str | None = Field(None, max_length=200, description="商户")
    registered_address: str | None = Field(None, max_length=500, description="注册地址")
    legal_person: str | None = Field(None, max_length=100, description="法人")
    previous_name: str | None = Field(None, max_length=200, description="曾用名")
    store_long_id: str | None = Field(None, max_length=100, description="store_long_id")
    store_short_id: str | None = Field(None, max_length=100, description="store_short_id")
    settlement_period: str | None = Field(None, max_length=100, description="settlement_period")
    primary_account: str | None = Field(None, max_length=200, description="primary_account")
    anchor: str | None = Field(None, max_length=100, description="主播")
    shop_type: str | None = Field(None, max_length=100, description="类型")
    purpose: str | None = Field(None, max_length=200, description="purpose")
    former_name: str | None = Field(None, max_length=200, description="former_name")


class ShopCreate(ShopInfoFields):
    platform_name: str = Field(..., min_length=1, max_length=50, description="平台")
    shop_name: str = Field(..., min_length=2, max_length=200, description="店铺名称")
    shop_color: str | None = Field(None, max_length=20, description="店铺颜色")
    remark: str | None = Field(None, max_length=2000, description="备注")


class ShopUpdate(ShopInfoFields):
    platform_name: str | None = Field(None, min_length=1, max_length=50, description="平台")
    shop_name: str | None = Field(None, min_length=2, max_length=200, description="店铺名称")
    shop_color: str | None = Field(None, max_length=20, description="店铺颜色")
    remark: str | None = Field(None, max_length=2000, description="备注")
    status: int | None = Field(None, ge=0, le=1, description="状态")


class ShopOut(ShopInfoFields):
    id: int
    org_id: int
    org_name: str | None = None
    platform_name: str
    shop_name: str
    shop_color: str | None
    remark: str | None
    status: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class ShopImportError(BaseModel):
    row: int
    message: str


class ShopImportResult(BaseModel):
    total: int
    created: int
    updated: int
    skipped: int
    errors: list[ShopImportError] = Field(default_factory=list)
